### PR TITLE
Dump edpm container images in separate log file

### PR DESCRIPTION
**Pull request by bshewale**
As a pull request owner and reviewers, I checked that:

Depends-On: https://github.com/openstack-k8s-operators/ci-framework/pull/424

- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: ci_framework/roles/edpm_build_images/tasks/package.yml [hunk 1]
```diff
@@ -24,3 +24,12 @@
       buildah bud -f ./Containerfile.ramdisk
       -t ironic-python-agent:{{ cifmw_edpm_build_images_tag }}
       --logfile {{ cifmw_edpm_build_images_basedir }}/logs/edpm_images/ironic_python_agent_container_package.log
+
+- name: Dump edpm container images in the file
+  args:
+    chdir: "{{ cifmw_edpm_image_builder_repo_path }}"
+  ansible.builtin.shell: |
+    sudo buildah images | grep "{{ item }}" | tee {{ cifmw_edpm_build_images_basedir }}/logs/containers-built.log
```
Comment (cjeanner) [inline ci_framework/roles/edpm_build_images/tasks/package.yml:32]: you probably want `tee -a` else it will override upon each iteration.
We may probably avoid iteration using a tweaked `| grep` such as `| grep -E '(edpm-hardened-uefi|ironic-python-agent)'` (NOT TESTED).

File: zuul.d/edpm_build_images.yaml [hunk 2]
```diff
@@ -25,3 +25,4 @@
       - ^ci/playbooks/edpm_build_images/.*
       - ^scenarios/centos-9/edpm_build_images_ci.yml
       - ^zuul.d/edpm_build_images.yaml
+      - ^ci_framework/roles/edpm_build_images/*
```
Comment (cjeanner) [inline zuul.d/edpm_build_images.yaml:28]: ah, good catch - that would avoid potential regressions!

File: ci_framework/roles/copy_container/files/copy-quay/utils.go [hunk 3]
```diff
@@ -189,6 +189,11 @@ func fetchLogs(URL string) string {
         data , _ = _fetch(path)
     }
 
+    if data == "" {
```
Comment (cjeanner) [inline ci_framework/roles/copy_container/files/copy-quay/utils.go:192]: why is this needed? Care to amend your commit message to explain why you have to modify some go code?

Comment (bshewale) [reply ci_framework/roles/copy_container/files/copy-quay/utils.go:192]: Hi Cedric, Quay push job basically checks the containers-built.log file and i am creating that fil in this PR but the path is different so i will need to add that here as well. I will add more info in the commit message once the job will green this PR is still in work in progress phase.

**Review by raukadah — APPROVED** [review]
passing noop https://review.rdoproject.org/zuul/build/36b67adc70af40dd99653b19758999a1 : SUCCESS in 0s
periodic-edpm-build-push-images-centos-9-antelope https://review.rdoproject.org/zuul/build/2cf01d4d59e14a9788782ed3d36be36a : SUCCESS in 16m 24s
periodic-edpm-container-image-quay-push-centos-9-antelope https://review.rdoproject.org/zuul/build/f01bb3a66eb24d4ea38b7c6208d7de88 : SUCCESS in 5m 32s
[5:43](https://redhat-internal.slack.com/archives/D04MDMHAJFN/p1690546430832839)
https://quay.io/repository/podified-antelope-centos9/edpm-hardened-uefi?tab=history

**Comment by raukadah** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Review by cjeanner — APPROVED** [review]
/approve

**Comment by bshewale** [comment]
recheck

**Comment by bshewale** [comment]
recheck

**Comment by raukadah** [comment]
recheck

**Comment by bshewale** [comment]
recheck

**Review by raukadah — APPROVED** [review]
/approve
